packages/docusaurus/src/commands/writeTranslations.ts

```diff
@@ -74,7 +74,9 @@ async function writePluginTranslationFiles({
 
 export async function writeTranslations(
   siteDir: string,
-  options: WriteTranslationsOptions & ConfigOptions & {locale?: string},
+  options: Partial<
```

Comment:
not fan of this kind of change 🤷‍♂️ I'd rather do the opposite and ensure all commands are never partial => handle edge cases and apply defaults at the edge of the system

Reply:
The problem is that `writeTranslations` should not be seen as fully internal. It's exported from the main interface and users may import `@docusaurus/core` and use it as a programmatic API. Therefore, I'd keep maximum compatibility with the CLI interface.

Reply:
Related: #4841 We should always assume the existence of those users that don't interact through the CLI.

Reply:
ok, that seems reasonable

What I don't like is to have long methods receiving partial options, where there's always a risk to end up using the partial option on which default was not properly applied

Can we split the methods in 2, something like this?

```ts
function doWriteTranslations(siteDir: string, options: Options) {
	// actual code
} 

export function writeTranslations(siteDir: string, options: Partial<Options>) {
	return doWriteTranslations(siteDir, applyDefaults(options))
} 
```

Reply:
Yes, makes sense.